### PR TITLE
[Braze Cloud] Adds sync mode to new createAlias V2 action

### PR DESCRIPTION
**Pull request by nick-Ag**
This PR creates a V2 version of the createAlias action which supports sync mode.

## Testing
[Testing Doc](https://docs.google.com/document/d/1Aq_kkOabFYItCCr-gFth2tO5ixUKoMaBjn0wGLRmCtw/edit?usp=sharing)

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/braze/createAliasV2.types.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Generated file. DO NOT MODIFY IT BY HAND.
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/braze/createAliasV2.types.ts:1]: This file needs to be removed

File: packages/destination-actions/src/destinations/braze/createAlias/index.ts
Comment (marinhero) [inline packages/destination-actions/src/destinations/braze/createAlias/index.ts:1]: This file does not require any updates

File: packages/destination-actions/src/destinations/braze/createAlias2/__tests__/snapshot.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    })
+
+    await expect(
+      testDestination.testAction(actionSlug, {
+        event: event,
+        mapping: { ...event.properties, __segment_internal_sync_mode: 'upsert' },
+        settings: settingsData,
+        auth: undefined
+      })
+    ).rejects.toThrowError()
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/braze/createAlias2/__tests__/snapshot.test.ts:27]: NIT: Make sure you are not swallowing other errors as this is a catch all for all the exceptions. It's better if you compare against the actual exception you are throwing.

File: packages/destination-actions/src/destinations/braze/createAlias2/__tests__/snapshot.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  })
+
+  it('required fields', async () => {
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/braze/createAlias2/__tests__/snapshot.test.ts:30]: NIT: be more descriptive with what the test is doing

**Review by marinhero — CHANGES_REQUESTED** [review]
Some of these are NITs but I do think that we should delete/reset the two files I flagged.

**Review by maryamsharif — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @nick-Ag PR deployed

**Pull request merged**